### PR TITLE
Newapi

### PR DESCRIPTION
**Pull request by crazy2be**
Basically moving everything off of `*Context` and onto the respective structs (i.e. `*Value`, `*Object`). Limited API changes outside of that.

@sqs: What are your thoughts? Basically just looking for a "LGTM" or "I think this should change."

### CODE REVIEW

File: errors.go [hunk 1]
```diff
@@ -53,5 +53,5 @@ func (r errorValue) Error() string {
 		panic("errorValue.ref is nil")
 	}
 	v := r.ctx.newValue(r.ref)
-	return r.ctx.ToStringOrDie(v)
+	return v.ToStringOrDie()
```
Comment (sqs) [inline errors.go:56]: Unrelated to this PR, but what about renaming this to ToStringOrPanic for clarity?

**Comment by sqs** [comment]
LGTM!

File: errors.go (hunk 1 above)
Comment (crazy2be) [inline errors.go:56]: Yeah, I was thinking about what to do with these. You could call them `MustToString()` (like the regexp package), or make a `.ToString()` that will never fail (and returns an empty string when the underlying method gives an error), along with a `.TryToString()` that returns an error. I think in most cases you really don't care about whether or not you actually got a "valid" value, and you would rather just keep on going.

Actually, otto's API for this is fairly nice- it always returns sane defaults, and an error. You can just ignore the error if you don't care about it: https://github.com/robertkrimen/otto

Comment (sqs) [inline errors.go:56]: Yeah, I like that idea. It's like strconv.ParseBool, etc., too.

**Pull request merged**